sdk/src/main/scala/com/horizen/SidechainMemoryPool.scala

```diff
@@ -129,14 +177,18 @@ class SidechainMemoryPool(unconfirmed: TrieMap[String, SidechainMemoryPoolEntry]
     }
 
     for (t <- txs) {
-      unconfirmed.put(t.id, SidechainMemoryPoolEntry(t))
+      val entry = SidechainMemoryPoolEntry(t)
+      if (entry.feeRate.getFeeRate() >= minFeeRate) {
+        addWithSizeCheck(entry)
+      }
     }
 
     this
   }
 
   override def remove(tx: SidechainTypes#SCBT): SidechainMemoryPool = {
     unconfirmed.remove(tx.id)
+    usedPoolSizeBytes = usedPoolSizeBytes - tx.size()
```

Comment:
Instead of using `tx.size()` we may, take the result of `unconfirmed.remove(tx.id)`.
If result is `Some(entry)` -> get entry.size() and decrease `usedPoolSizeBytes`.
In case we tried to remove nonexistent tx we don't decrease  `usedPoolSizeBytes`